docs/content/operations/api-reference.md

```diff
@@ -412,10 +412,18 @@ druid.query.segmentMetadata.defaultHistory
 
 Returns the dimensions of the datasource.
 
+<div class="note caution">
```

Comment:
Maybe include comments on what to use instead would be better.

Reply:
segmentMetadata queries or Druid SQL INFORMATION_SCHEMA, probably. And I would include the reason: they provide more comprehensive information, and they work on streaming datasources too, which these APIs do not always.

Reply:
Added.